pilot/cmd/pilot-agent/status/server.go

```diff
@@ -55,17 +65,34 @@ type Server struct {
 	ready               *ready.Probe
 	mutex               sync.RWMutex
 	lastProbeSuccessful bool
+	appKubeProbers      KubeAppProbers
 }
 
 // NewServer creates a new status server.
-func NewServer(config Config) *Server {
-	return &Server{
+func NewServer(config Config) (*Server, error) {
+	s := &Server{
 		statusPort: config.StatusPort,
 		ready: &ready.Probe{
 			AdminPort:        config.AdminPort,
 			ApplicationPorts: config.ApplicationPorts,
 		},
 	}
+	if config.KubeAppHTTPProbers == "" {
+		return s, nil
+	}
+	if err := json.Unmarshal([]byte(config.KubeAppHTTPProbers), &s.appKubeProbers); err != nil {
+		return nil, fmt.Errorf("failed to decode app http prober err = %v, json string = %v", err, config.KubeAppHTTPProbers)
+	}
+	// Validate the map key matching the regex pattern.
+	for path, prober := range s.appKubeProbers {
+		if !appProberPattern.Match([]byte(path)) {
+			return nil, fmt.Errorf("invalid key, must be in form of /app-health/container-name/ready|live")
```

Comment:
`^/app-health/[^\/]+/(livez|readyz)$`  is this right?

Reply:
Okay, you mean put the actual matching regex string here. Sure.